File: surgicalprep-backend/app/schemas/instrument.py
"""
Pydantic schemas for instrument-related requests and responses.
"""
from datetime import datetime
from typing import List, Optional
from pydantic import BaseModel, Field


class InstrumentBase(BaseModel):
    name: str = Field(..., max_length=200)
    aliases: Optional[List[str]] = None
    category: str = Field(..., max_length=100)
    description: Optional[str] = None
    primary_uses: Optional[List[str]] = None
    common_procedures: Optional[List[str]] = None
    handling_notes: Optional[str] = None
    image_url: Optional[str] = None
    thumbnail_url: Optional[str] = None
    is_premium: bool = False


class InstrumentCreate(InstrumentBase):
    pass


class InstrumentUpdate(BaseModel):
    name: Optional[str] = Field(default=None, max_length=200)
    aliases: Optional[List[str]] = None
    category: Optional[str] = Field(default=None, max_length=100)
    description: Optional[str] = None
    primary_uses: Optional[List[str]] = None
    common_procedures: Optional[List[str]] = None
    handling_notes: Optional[str] = None
    image_url: Optional[str] = None
    thumbnail_url: Optional[str] = None
    is_premium: Optional[bool] = None


class InstrumentResponse(InstrumentBase):
    id: str
    created_at: datetime
    updated_at: datetime
    
    class Config:
        from_attributes = True


class InstrumentListResponse(BaseModel):
    id: str
    name: str
    category: str
    thumbnail_url: Optional[str] = None
    is_premium: bool
    
    class Config:
        from_attributes = True


class InstrumentSearchParams(BaseModel):
    query: Optional[str] = None
    category: Optional[str] = None
    page: int = Field(default=1, ge=1)
    page_size: int = Field(default=20, ge=1, le=100)


class PaginatedInstruments(BaseModel):
    items: List[InstrumentListResponse]
    total: int
    page: int
    page_size: int
    total_pages: int


# Category list
class CategoryResponse(BaseModel):
    name: str
    count: int

File: surgicalprep-mobile/src/types/instrument.ts
// Instrument types matching backend Pydantic schemas

export type InstrumentCategory =
  | 'cutting'
  | 'clamping'
  | 'grasping'
  | 'retracting'
  | 'suturing'
  | 'suctioning'
  | 'dilating'
  | 'probing'
  | 'measuring'
  | 'specialty';

export interface Instrument {
  id: string;
  name: string;
  aliases: string[];
  category: InstrumentCategory;
  description: string;
  primary_uses: string[];
  common_procedures: string[];
  handling_notes: string | null;
  image_url: string | null;
  thumbnail_url: string | null;
  is_premium: boolean;
  created_at: string;
  updated_at: string;
}

export interface InstrumentListParams {
  search?: string;
  category?: InstrumentCategory;
  page?: number;
  limit?: number;
}

export interface InstrumentListResponse {
  items: Instrument[];
  total: number;
  page: number;
  page_size: number;
  total_pages: number;
}

export interface InstrumentBookmark {
  instrument_id: string;
  bookmarked_at: string;
}

// Category display configuration
export const CATEGORY_CONFIG: Record<InstrumentCategory, { label: string; color: string; bgColor: string }> = {
  cutting: { label: 'Cutting', color: '#DC2626', bgColor: '#FEE2E2' },
  clamping: { label: 'Clamping', color: '#7C3AED', bgColor: '#EDE9FE' },
  grasping: { label: 'Grasping', color: '#2563EB', bgColor: '#DBEAFE' },
  retracting: { label: 'Retracting', color: '#059669', bgColor: '#D1FAE5' },
  suturing: { label: 'Suturing', color: '#D97706', bgColor: '#FEF3C7' },
  suctioning: { label: 'Suctioning', color: '#0891B2', bgColor: '#CFFAFE' },
  dilating: { label: 'Dilating', color: '#DB2777', bgColor: '#FCE7F3' },
  probing: { label: 'Probing', color: '#4F46E5', bgColor: '#E0E7FF' },
  measuring: { label: 'Measuring', color: '#65A30D', bgColor: '#ECFCCB' },
  specialty: { label: 'Specialty', color: '#6B7280', bgColor: '#F3F4F6' },
};
